libdnf/utils/utils.cpp

```diff
@@ -123,18 +123,16 @@ std::string trimPrefix(const std::string &source, const std::string &prefix)
     throw std::runtime_error{"Prefix '" + prefix + "' not found"};
 }
 
-bool startsWith(const std::string &source, const std::string &toMatch)
+bool startsWith(const std::string & source, const std::string & toMatch)
 {
-    return source.find(toMatch) == 0;
+    return strncmp(source.c_str(), toMatch.c_str(), toMatch.size()) == 0;
 }
 
-bool endsWith(const std::string &source, const std::string &toMatch)
+bool endsWith(const std::string & source, const std::string & toMatch)
 {
-    auto it = toMatch.begin();
-    return source.size() >= toMatch.size() &&
-           std::all_of(std::next(source.begin(), source.size() - toMatch.size()), source.end(), [&it](const char &c) {
-               return c == *(it++);
-           });
+    auto toMatchSize = toMatch.size();
+    return source.size() >= toMatchSize &&
+    !strcmp(source.c_str() + source.size() - toMatchSize, toMatch.c_str());
```

Comment:
I'd expect an additional indent on this line, as its a continuation of the previous one.

Also, why not use `std::string::compare` (for both functions)?

Reply:
Please @lukash could you explain me what is the benefit of using std::string::compare?

Reply:
It's a C++ approach in C++ code using the designated `std::string` methods. Its IMO the idiomatic way for C++. Functionally no difference, performance-wise I'd expect they'd be close to the same.